make/tools.mk

```diff
@@ -10,6 +10,8 @@ export PATH := $(PWD)/bin/tools:$(PATH)
 
 CTR=docker
 
+CMREL=$(PWD)/bin/tools/cmrel
```

Comment:
I'd suggest adding a comment explaining why this is the only variable, and why we don't have others (e.g., YQ, etc.)

```suggestion
# We usually use the relative path to binaries (e.g., bin/tools/crane) because it matches the path that we have
# to use when using that binary as a prerequisite in Makefile targets. That makes the process of visually
# checking that the tool used in a target is also in the prerequisites. In the case of cmrel, one of the targets
# require the absolute path of the tool, so we have a variable for it.
CMREL=$(PWD)/bin/tools/cmrel
```

Reply:
The reason I started removing CMREL and the other variables in https://github.com/cert-manager/cert-manager/pull/4988 is because it makes it easier to spot when a tool is used but isn't in the prerequisite list. For example, in the following example, we are referring to cosign in two different ways: bin/tools/cosign and $(COSIGN):

```sh
bin/SHA256SUMS.sig: bin/SHA256SUMS bin/tools/cosign
	$(COSIGN) sign-blob --key $(COSIGN_KEY) $< > $@
```

A clearer way is to use `bin/tools/cosign` in both cases:

```sh
bin/SHA256SUMS.sig: bin/SHA256SUMS bin/tools/cosign
	bin/tools/cosign sign-blob --key $(COSIGN_KEY) $< > $@
```

But that's a very subjective argument 😇

Reply:
I realize my idea of making it "visually easier to spot when a tool is used but isn't in the prerequisite list" is a bit stupid. I am happy to revert back to using variables everywhere.

Reply:
> make it "visually easier to spot when a tool is used but isn't in the prerequisite list" is very subjective

I agree it's subjective, but I also don't really feel it's a goal I'd aim for because if a tool is missing from the prereq list then it should be pretty obvious when building from a fresh start because if the tool isn't in the prereq list then it won't be provisioned and the tool won't exist.

I think having the variables makes it easy to sub in a different version of the tool which was the aim. Same reason we have `CTR=docker` but it's easy to change to podman or whatever.

I don't think we should shave this yak before the release of 1.8. We should minimise changes at this point. Let's focus on stability and getting the release out of the door. We can revisit later if we want to. It's not a big deal!

I'll incorporate your suggested comment :+1:

Reply:
Thanks. I had lost sight of the goal "make it easy to substitute in a different version of the tool".